Spec/Tests/RealtimeClientChannelTests.swift

```diff
@@ -3889,6 +3889,57 @@ class RealtimeClientChannelTests: XCTestCase {
             channel.attach()
         }
     }
+    
+    // RTL13b
+    func test__131b__Channel__if_the_channel_receives_a_server_initiated_DETACHED_message_and_if_the_attempt_to_reattach_fails_then_the_channel_will_transition_to_SUSPENDED_state_with_periodic_reattach_with_channelRetryTimeout() {
+        
+        let options = AblyTests.commonAppSetup()
+        options.channelRetryTimeout = 1.0
+        options.autoConnect = false
+        let client = ARTRealtime(options: options)
+        client.internal.setTransport(TestProxyTransport.self)
+        
+        let previousRealtimeRequestTimeout = ARTDefault.realtimeRequestTimeout()
+        ARTDefault.setRealtimeRequestTimeout(1.0)
+        client.connect()
+        
+        defer { ARTDefault.setRealtimeRequestTimeout(previousRealtimeRequestTimeout); client.dispose(); client.close() }
+        
+        let transport = client.internal.transport as! TestProxyTransport
+        transport.actionsIgnored += [.attached]
+
+        expect(client.connection.state).toEventually(equal(ARTRealtimeConnectionState.connected), timeout: testTimeout)
+
+        let channel = client.channels.get(uniqueChannelName())
+        channel.attach()
+        
+        expect(channel.state).to(equal(ARTRealtimeChannelState.attaching))
+
+        let detachedMessageWithError = AblyTests.newErrorProtocolMessage()
+        detachedMessageWithError.action = .detached
+        detachedMessageWithError.channel = channel.name
+        
+        var retryNumber = 1
+        let maxRetryCount = 5
```

Comment:
Could we call this `numberOfRetriesToWaitFor` and put a comment saying that it's an arbitrary number chosen to give us confidence that the retries are continuing indefinitely? When I first read it I thought that 5 had some special meaning.

Reply:
https://github.com/ably/ably-cocoa/pull/1551/commits/4fa70650b6f05265b682e767a027646daa48fdb3